app/src/test/java/org/simple/clinic/allpatientsinfacility/AllPatientsInFacilityUiChangeProducerTest.kt

```diff
@@ -0,0 +1,92 @@
+package org.simple.clinic.allpatientsinfacility
+
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.verify
+import com.nhaarman.mockito_kotlin.verifyNoMoreInteractions
+import com.nhaarman.mockito_kotlin.verifyZeroInteractions
+import io.reactivex.disposables.Disposable
+import io.reactivex.subjects.PublishSubject
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.simple.clinic.patient.PatientMocker
+import java.util.UUID
+
+class AllPatientsInFacilityUiChangeProducerTest {
+  private val view = mock<AllPatientsInFacilityView>()
+  private val uiChangeProducer = AllPatientsInFacilityUiChangeProducer()
+  private val viewStateSubject = PublishSubject.create<AllPatientsInFacilityViewState>()
+
+  private val facility = PatientMocker.facility(UUID.fromString("1be5097b-1c9f-4f78-aa70-9b907f241669"))
+
+  private lateinit var disposable: Disposable
+
+  @Before
+  fun setUp() {
+    disposable = viewStateSubject
+        .compose(uiChangeProducer)
+        .subscribe { uiChange -> uiChange(view) }
+  }
+
+  @After
+  fun tearDown() {
+    disposable.dispose()
+  }
+
+  @Test
+  fun `when a facility is being fetched, then do nothing`() {
+    // when
+    viewStateSubject.onNext(AllPatientsInFacilityViewState.FETCHING_PATIENTS_STATE)
+
+    // then
+    verifyZeroInteractions(view)
+  }
+
+  @Test
+  fun `when a facility is fetched, then do nothing`() {
+    // given
+    val fetchingPatientsState = AllPatientsInFacilityViewState
+        .FETCHING_PATIENTS_STATE
```

Comment:
I think `AllPatientsInFacilityViewState.FETCHING_PATIENTS_STATE` can be extracted to a variable which is an initial state. 
Or we could rename `FETCHING_PATIENTS_STATE` to `FETCHING_FACILITIES`

Reply:
Good point. Fixed.